src/globus_sdk/services/search/data.py

```diff
@@ -221,6 +224,49 @@ def add_sort(
         return self
 
 
+class SearchQueryV1(utils.PayloadWrapper):
+    """
+    A specialized dict which has helpers for creating and modifying a Search
+    Query document. Replaces the usage of ``SearchQuery``.
+
+    :param q: The query string. Required unless filters are used.
+    :param limit: A limit on the number of results returned in a single page
+    :param offset: An offset into the set of all results for the query
+    :param advanced: Whether to enable (``True``) or not to enable (``False``) advanced
+        parsing of query strings. The default of ``False`` is robust and guarantees that
+        the query will not error with "bad query string" errors
+    :param additional_fields: additional data to include in the query document
+    """
+
+    def __init__(
+        self,
+        q: str | None = None,
+        *,
```

Comment:
I'm not sure about this, but I think we want to move primarily to keyword-only arguments.

With a named positional, you're actually declaring two supported usages: `SearchQueryV1("*")` and `SearchQueryV1(q="*")`. Typically it's fine (we lived with it until Python 3, after all), until the following two things happen:
- you add a second named-positional argument
- you deprecate or remove the first named-positional argument

After such changes to an API, positional usage becomes problematic. So then you're in the position of only supporting keyword-arg usage anyway.

In Python APIs, I've increasingly tended towards keyword-only arguments:

```suggestion
        *,
        q: str | None = None,
```

One thing about this is that it's a reversible decision, whereas declaring support for a positional arg is not.
Once `SearchQueryV1("*")` is supported, support must be maintained.
But if `SearchQueryV1(q="*")` is the only supported usage, it can easily be updated to also allow `SearchQueryV1("*")` in the future.